### ISSUE TITLE
DynamoDB Connection failing with "__init__() got an unexpected keyword argument 'min_insert_date'"

### ISSUE THREAD

**Issue by lucleray** [getredash/redash#1748]
Hello,

I have an issue connecting Redash to my DynamoDB table in AWS.

### Issue Summary

I have the follow issue when I test my DynamoDB connection :

`Error running query: __init__() got an unexpected keyword argument 'min_insert_date'`

<img width="314" alt="capture d ecran 2017-05-03 a 18 28 56" src="https://cloud.githubusercontent.com/assets/6616955/25657054/7ff794c0-302e-11e7-8797-8f4fd65aba68.png">

If I go to the logs of the worker, I have the following : 
```
[2017-05-03 10:17:11,169: INFO/Worker-15] redash.tasks.execute_query[524b9ba4-1b53-47fc-95e3-345b6f6ee765]: task=execute_query state=executing_query query_hash=15dd79f96b76332ec149298a920e9d97 type=dynamodb_sql ds_id=1  task_id=524b9ba4-1b53-47fc-95e3-345b6f6ee765 queue=queries query_id=adhoc username=luc@supahands.com
worker_1    | [2017-05-03 10:17:11,217: WARNING/Worker-15] Unexpected error while running query:
worker_1    | Traceback (most recent call last):
worker_1    |   File "/app/redash/tasks/queries.py", line 421, in run
worker_1    |     data, error = query_runner.run_query(annotated_query, self.user)
worker_1    |   File "/app/redash/query_runner/dynamodb_sql.py", line 98, in run_query
worker_1    |     engine = self._connect()
worker_1    |   File "/app/redash/query_runner/dynamodb_sql.py", line 85, in _connect
worker_1    |     engine.connect(**config)
worker_1    |   File "/usr/local/lib/python2.7/dist-packages/dql/engine.py", line 114, in connect
worker_1    |     self.connection = DynamoDBConnection.connect(*args, **kwargs)
worker_1    |   File "/usr/local/lib/python2.7/dist-packages/dynamo3/connection.py", line 194, in connect
worker_1    |     return cls(client, **kwargs)
worker_1    | TypeError: __init__() got an unexpected keyword argument 'min_insert_date'
```

- [x] I checked that all the python and node dependencies were installed and up-to-date.
- [x] I tried to connect to another database (mongodb) and it worked fine.

### Steps to Reproduce

1. I added a data source, configured the connection, entered the access/secret keys and region.

2. I clicked on the **Test Connection** button 

### Technical details:

* Redash Version: 1.0.3
* Browser/OS: Chrome/OS X 10.11.6
* How did you install Redash: with Docker, following this tutorial : 
   https://redash.io/help-onpremise/dev/docker.html

**Comment by arikfr** [comment]
This isn't an issue with DynamoDB, but a UI issue -- when changing the data source type, it wasn't cleaning the properties, so the connection options get unknown additional properties.

Fixed in #1754.

**Comment by lucleray** [comment]
I still have an issue on the UI.

In chrome's console :
```
TypeError: Cannot read property 'configuration_schema' of undefined
    at s (dynamic-form.js:37)
    at dynamic-form.js:115
    at p.$digest (angular.js:17524)
    at p.$apply (angular.js:17790)
    at l (angular.js:11831)
    at b (angular.js:12033)
    at XMLHttpRequest.k.onload (angular.js:11966)
```
